src/test/java/ca/ryangreen/apigateway/generic/GenericApiGatewayClientTest.java

```diff
@@ -34,6 +38,9 @@
 public class GenericApiGatewayClientTest {
```

Comment:
Please ensure we have the following test cases:
+ 0 query params
+ 1 query param
+ > 1 query params
+ query parameters with characters that need to be url encoded